File: src/main/java/ru/citeck/idea/references/resolvers/JSLiteralExpressionValueResolver.java
package ru.citeck.idea.references.resolvers;

import com.intellij.lang.javascript.psi.JSLiteralExpression;
import ru.citeck.idea.references.PsiElementValueResolver;

public class JSLiteralExpressionValueResolver implements PsiElementValueResolver<JSLiteralExpression> {

    @Override
    public Class<JSLiteralExpression> getPsiElementType() {
        return JSLiteralExpression.class;
    }

    @Override
    public Object getValue(JSLiteralExpression psiElement) {
        return psiElement.getStringValue();
    }
}

File: src/main/kotlin/ru/citeck/idea/references/resolvers/JSLiteralExpressionValueResolver.kt
package ru.citeck.idea.references.resolvers

import com.intellij.lang.javascript.psi.JSLiteralExpression
import ru.citeck.idea.references.PsiElementValueResolver

class JSLiteralExpressionValueResolver : PsiElementValueResolver<JSLiteralExpression> {

    override fun getPsiElementType(): Class<JSLiteralExpression> {
        return JSLiteralExpression::class.java
    }

    override fun getValue(psiElement: JSLiteralExpression): Any? {
        return psiElement.stringValue
    }
}
